week1/homework/src/server.js

```diff
@@ -10,6 +10,35 @@ function createServer(port) {
 
   const server = http.createServer((request, response) => {
     // TODO: Write your homework code here
+    switch (request.url) {
+      case '/state':
+        response.setHeader('Content-Type', 'application/json');
+        response.write(JSON.stringify({ state: state }));
+        break;
+
+      case '/add':
+        state += 1;
+        response.setHeader('Content-Type', 'application/json');
+        response.write(JSON.stringify({ state: state}));
+
+        break;
+      case '/subtract':
+        state -= 1;
+        response.setHeader('Content-Type', 'application/json');
+        response.write(JSON.stringify({ state: state}));
```

Comment:
💡 PRO TIP: If you find yourself repeating a piece of code several times perhaps it is a good idea to put it in a separate function instead of copy-pasting it